pennylane/vqe/vqe.py

```diff
@@ -73,54 +79,57 @@ class Hamiltonian:
     :doc:`/introduction/chemistry` module can be used to generate a molecular
     Hamiltonian.
 
-    .. Warning::
-
-        Hamiltonians can be constructed using Pythonic arithmetic operations. For example:
+    Hamiltonians can alternatively be constructed using Pythonic arithmetic operations. For example:
 
-        >>> qml.PauliX(0) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
+    >>> qml.PauliX(0) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
 
-        is equivalent to the following Hamiltonian:
+    is equivalent to the following Hamiltonian:
 
-        >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
+    >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
 
-        When Hamiltonians are defined using arithmetic operations **inside of QNodes**, constituent observables
-        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside of QNodes,
-        or inside of QNodes using the conventional method.
+    .. Warning::
+        When Hamiltonians are defined using arithmetic operations **inside of quantum functions**, constituent observables
+        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside
+        of quantum functions, or inside of quantum functions using the conventional construction method.
 
-        Note that this issue also arises when calling the ``simplify()`` method.
     """
-    # Todo: this is a temporary solution to make the circuit drawer work
-    num_params = 0
 
-    def __init__(self, coeffs, observables, simplify=False):
+    num_wires = qml.operation.AnyWires
+    num_params = 1
+    par_domain = "A"
+    grad_method = "A"  # supports analytic gradients
 
-        if len(coeffs) != len(observables):
+    def __init__(self, coeffs, observables, simplify=False, id=None, do_queue=True):
+
+        if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        if any(np.imag(coeffs) != 0):
-            raise ValueError(
-                "Could not create valid Hamiltonian; " "coefficients are not real-valued."
-            )
-
```

Comment:
Removed this, it is too micro-managed...

Reply:
🎉 We've had a lot of users asking for this restriction to be lifted!